src/libraries/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Directory.Changed.cs

```diff
@@ -75,8 +78,26 @@ public void FileSystemWatcher_Directory_Changed_SymLink()
                 Action action = () => File.AppendAllText(file.Path, "longtext");
                 Action cleanup = () => File.AppendAllText(file.Path, "short");
 
-                ExpectEvent(watcher, 0, action, cleanup, dir.Path);
+                ExpectEvent(watcher, 0, action, cleanup, expectedPath: null);
+            }
+        }
+
+        private string GetDirectoryOrSymLink(string pathToDirectory)
+        {
+            if (!UseSymLink)
+            {
+                return pathToDirectory;
             }
+
+            string symLinkPath = GetTestFilePath();
+            Assert.True(CreateSymLink(targetPath: pathToDirectory, linkPath: symLinkPath, isDirectory: true));
+            return symLinkPath;
         }
     }
+
+    [ConditionalClass(typeof(Symlink_Directory_Changed_Tests), nameof(CanCreateSymbolicLinks))]
+    public class Symlink_Directory_Changed_Tests : Directory_Changed_Tests
+    {
+        public override bool UseSymLink => true;
+    }
```

Comment:
Consider moving this class to its own file.

Reply:
I was planning on expanding this pattern to the other `Directory_*_Tests` classes, with that said, do you think that adding one file for each `Symlink_Directory_*_Tests` is worth? IMO is more clear to just keep it in the same file.

An alternative could be having one single file for all the symlink test classes.

Reply:
Either option is fine. Consider my suggestion optional.